compute_tools/src/http/api.rs

```diff
@@ -125,13 +125,23 @@ async fn routes(req: Request<Body>, compute: &Arc<ComputeNode>) -> Response<Body
         (&Method::POST, route) if route.starts_with("/extension_server/") => {
             info!("serving {:?} POST request", route);
             info!("req.uri {:?}", req.uri());
+
```

Comment:
If this download takes a long time, there will be timeouts in production. This means that the downloads will get interrupted. I am unsure if a download is first done to a tempfile and then renamed as the real target file name, but it's something to consider.

If timeouts are possible, then it's possible that you have a duplicate calls to download the same thing.

It might be possible to use the "downloaded extensions" as a map to avoid launching two downloads at the same time (probably requires help of `Once`). We don't have a good example of this in pageserver.

Reply:
We will continue to look into Joonas' review comments 
Merged this branch into my other branch just to make it more possible to understand what is going on in git